packages/qwik/src/core/v2/shared/jsx-props.ts

```diff
@@ -0,0 +1,34 @@
+import type { JSXNode } from '@builder.io/qwik/jsx-runtime';
+import { _IMMUTABLE } from '../../internal';
+
+export function fixJsxProps(jsx: JSXNode<unknown>) {
```

Comment:
So this function scans both objects and does a whole bunch of megamorphic reads.

Can you explain why is it needed? Are there other faster ways to do this?

Reply:
so basically sometimes mutable props are immutable and we need fix that
 the code is from here
 https://github.com/BuilderIO/qwik/blob/cf86daa15f62b429b5b25be83ccda9b40a4d3d7c/packages/qwik/src/core/render/ssr/render-ssr.ts#L602